projects/core/src/product/event/product-event.builder.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { filter, map, pairwise } from 'rxjs/operators';
+import { EventService } from '../../event/event.service';
+import { Breadcrumb } from '../../model/product-search.model';
+import { createFrom } from '../../util/create-from';
+import { ProductSearchService } from '../facade';
+import { FacetChangedEvent } from './product.events';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductEventBuilder {
+  constructor(
+    protected eventService: EventService,
+    protected productSearchService: ProductSearchService
+  ) {
+    this.register();
+  }
+
+  protected register() {
+    this.eventService.register(
+      FacetChangedEvent,
+      this.buildFacetChangedEvent()
+    );
+  }
+
+  protected buildFacetChangedEvent(): Observable<FacetChangedEvent> {
```

Comment:
Just a note, I would see value in adding some JSDoc to the functions and maybe also some small comments to explain the more complex parts like "isCategory".